pkg/sql/catalog/descs/txn.go

```diff
@@ -89,6 +89,66 @@ type TxnWithExecutorFunc = func(
 	ie sqlutil.InternalExecutor,
 ) error
 
+// TxnDB include methods to run transactions with a *Collection.
+type TxnDB interface {
```

Comment:
Do we really want this? Wouldn't it be enough to inject `sqlutil.InternalExecutorFactory` and use these methods directly?